src/main/java/com/epam/aidial/core/Proxy.java

```diff
@@ -121,12 +122,19 @@ private void handleError(Throwable error, HttpServerRequest request) {
      * Called when proxy received the request headers from a client.
      */
     private void handleRequest(HttpServerRequest request) {
+        enableCors(request);
```

Comment:
I think that's not a full implementation of CORS handler. Please take a look at a spec https://developer.mozilla.org/en-US/docs/Glossary/Preflight_request#

You can review the implementation in vertex.io io.netty.handler.codec.http.cors.CorsHandler

Reply:
The HTTP code 204 doesn't seem to be necessary. Currently, we do not use any successful code other than 200 in the application. Additionally, the Spring implementation of CORS also returns code 200 for OPTIONS requests. I'd keep it similar for now.